driver-core/src/main/com/mongodb/internal/connection/ClusterableServerFactory.java

```diff
@@ -18,11 +18,10 @@
 
 import com.mongodb.ServerAddress;
 import com.mongodb.connection.ServerSettings;
-import com.mongodb.event.ServerListener;
 
 public interface ClusterableServerFactory {
     ClusterableServer create(ServerAddress serverAddress, ServerDescriptionChangedListener serverDescriptionChangedListener,
-                             ServerListener serverListener, ClusterClock clusterClock);
```

Comment:
There is no need to pass the server listener here, and doing so made it more difficult to wrap the listener.  So removed the listener from this interface and made it the responsibility of implementers of this interface to have the appropriate (wrapped) listener available.